### ISSUE TITLE
Struggles with auto-sign in

### ISSUE THREAD

**Issue by joebelmonte** [ga-wdi-boston/game-project#659]
I'm working on a feature to automatically sign the user in when they sign up.  I have an auto sign in function that creates the account and then signs in on the click of the create account button.

It's not working, and from what I can see in the log I think the problem is that it attempts to sign in before the account has been created.  Is there a way I can force the sign in function to fire only after the sign up post has completed?

**Comment by MicFin** [comment]
Have you met all other requirements and doing this as a bonus or nice to have?  If not, please move this feature to the very end.

**Comment by joebelmonte** [comment]
I have met all other requirements other than removing the console logs and creating the readme.  Functionally, everything else works.  I'm about to give up on this but I figure it might be a quick fix.

**Comment by joebelmonte** [comment]
The solution was to nest the auto-sign in function as a .then for the sign up function:

```js
const onSignUp = function (event) {
  const data = getFormFields(this)
  event.preventDefault()
  api.signUp(data)
    .then(ui.signUpSuccess)
    .then(() => {
      api.signInAuto(data)
        .then(ui.autoSignInSuccess)
        .catch(ui.autoSignInFailure)
    })
    .catch(ui.signUpFailure)
}
```

Thanks, Mike!
